lib/spack/spack/test/versions.py

```diff
@@ -904,6 +904,7 @@ def test_version_list_normalization():
     assert ver("=1.0,ref=1.0,1.0:2.0") == ver(["1.0:2.0"])
     assert ver("=1.0,1.0:2.0,ref=1.0") == ver(["1.0:2.0"])
     assert ver("1.0:2.0,=1.0,ref=1.0") == ver(["1.0:2.0"])
+    assert ver("1.0:2.0,2.1,2.2:3,4:6") == ver(["1.0:6"])
```

Comment:
Can we add a brief comment above this use case, to explain that the we don't have intersection but the union is not disjoint?